config/initializers/redis.rb

```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
+
+require 'redis'
+
+## Added rescue condition if Redis connection is failed
+PORT = 6379
+
+begin
+  $redis = Redis.new(:host => Rails.configuration.redis_host, :port => PORT)
```

Comment:
Sharing the same Redis instance between threads may lead to tons of locking as specified here: https://github.com/redis/redis-rb/commit/9745e22db65ac294be51ed393b584c0f8b72ae98 but this solution is good enough for our current case.